ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1043,168 @@ def test_server_option_with_unreachable_ad(self):
+            group = "groups={0}(testgroup1@{1})".format(gid, self.ad_domain)
+            assert (gid_str in test_group.stdout_text
+                    and group in test_group.stdout_text)
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+    num_ad_subdomains = 0
+    num_ad_treedomains = 0
+    uid_override = "99999999"
+    gid_override = "78878787"
+
+    def test_add_posix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust-posix'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_corresponding_existing_group(self, type):
+        """
+        Test checks that sssd can resolve AD users which
+        contain posix attributes (uidNumber and gidNumber)
+        but there is no group with the corresponding gidNumber
+        and override ad trusted user in default trust view.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        with self.set_idoverrideuser(posixuser, uid="10060",
+                                     gid="10049"):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert (uid == "10060" and gid == "10049")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_only_uid_number_auto_private_group_default(self, type):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, auto-private-group
+        is set to false/true/hybrid
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        if type == "true":
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert uid == gid
+        else:
+            for host in [self.master, self.clients[0]]:
+                result = host.run_command(['id', posixuser], raiseonerr=False)
+                tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_group_primary_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) and there is primary group
+        with gid number defined.
+        """
+        posixuser = "testuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        assert uid == '10042'
+        if type == "true":
+            assert uid == gid
+        else:
+            assert gid == '10047'
+
```

Comment:
You can also add the following check:
in all the cases, the list of groups must also contain the user's group 10047(testgroup@ad.test)